src/commonMain/kotlin/app/cash/turbine/flow.kt

```diff
@@ -195,7 +195,8 @@ private fun <T> testInInternal(flow: Flow<T>, timeout: Duration?, scope: Corouti
     if (debug) println("Scope ending ${exception ?: ""}")
 
     // Only validate events were consumed if the scope is exiting normally.
-    if (exception == null) {
+    // CancellationException also indicates _normal_ cancellation of a coroutine.
+    if (exception == null || exception is CancellationException) {
```

Comment:
I think we can actually get rid of this whole `invokeOnCompletion`; `turbineScope` now has the code to run this validation, so we don't need it here. I'm kinda surprised it works, actually

Does not need to be changed before merge, I'll clean it up in a followup PR